exonum-java-binding/core/rust/src/storage/proof_list_index.rs

```diff
@@ -203,32 +205,43 @@ pub extern "system" fn Java_com_exonum_binding_core_storage_indices_ProofListInd
     utils::unwrap_exc_or(&env, res, ptr::null_mut())
 }
 
-/// Returns Java representation of the proof that an element exists at the specified index.
+/// Returns the proof that an element exists at the specified index. The proof is serialized in
+/// the protobuf format.
 #[no_mangle]
-// fixme: ECR-3614
-#[allow(unused_variables)]
 pub extern "system" fn Java_com_exonum_binding_core_storage_indices_ProofListIndexProxy_nativeGetProof(
     env: JNIEnv,
     _: JObject,
     list_handle: Handle,
     index: jlong,
-) -> jobject {
-    ptr::null_mut()
+) -> jbyteArray {
+    let res = panic::catch_unwind(|| {
+        let proof = match *handle::cast_handle::<IndexType>(list_handle) {
+            IndexType::SnapshotIndex(ref list) => list.get_proof(index as u64),
+            IndexType::ForkIndex(ref list) => list.get_proof(index as u64),
+        };
+        env.byte_array_from_slice(&proof.to_pb().write_to_bytes().unwrap())
+    });
+    utils::unwrap_exc_or(&env, res, ptr::null_mut())
 }
 
-/// Returns Java representation of the proof that some elements exists in the specified range.
+/// Returns the proof that some elements exists in the specified range. The proof is serialized in
+/// the protobuf format.
 #[no_mangle]
-// fixme: ECR-3614
-#[allow(unused_variables)]
 pub extern "system" fn Java_com_exonum_binding_core_storage_indices_ProofListIndexProxy_nativeGetRangeProof(
     env: JNIEnv,
     _: JObject,
     list_handle: Handle,
     from: jlong,
     to: jlong,
-) -> jobject {
-    // fixme: ECR-3614
-    ptr::null_mut()
+) -> jbyteArray {
+    let res = panic::catch_unwind(|| {
+        let proof = match *handle::cast_handle::<IndexType>(list_handle) {
+            IndexType::SnapshotIndex(ref list) => list.get_range_proof(from as u64..to as u64),
+            IndexType::ForkIndex(ref list) => list.get_range_proof(from as u64..to as u64),
+        };
+        env.byte_array_from_slice(&proof.to_pb().write_to_bytes().unwrap())
```

Comment:
It can be extracted in a function: `proof_to_bytes(env, proof) -> jbytearray`

Reply:
Done